GH-672: review 5

bertllll opened the pull request.

kauri-hero commented:
this is still a draft PR @masqrauder  ?
There are a lot of commits from earlier in year - looks like dependencies updates?

dnwiebe commented on .github/workflows/ci-matrix.yml:
How does this work?

dnwiebe commented on .idea/scopes/MASQ_Copyright.xml:
Really? This looks like it maybe got a little out of hand. It also looks a bit too specific. Can a lot of those explicit filenames be replaced with more generic wildcards?

dnwiebe commented on README.md:
To make the README file show the status of the build?

dnwiebe commented on masq/ci/run_integration_tests.sh:
Are you _sure_ this can be removed? I don't remember exactly why I put it in, but I do remember specifically putting it in. I'll bet there are race conditions somewhere in those tests. We may have put in tools recently to deal with them, but my guess is that we're not using them yet in the integration tests.

masqrauder commented on .idea/scopes/MASQ_Copyright.xml:
The IDE must've generated this. I didn't write it. It can be deleted.

masqrauder commented on README.md:
Yes. Front page shows build status with this.

masqrauder commented on masq/ci/run_integration_tests.sh:
Undone/reverted.

masqrauder commented on .github/workflows/ci-matrix.yml:
See the Summary tab at this link: https://github.com/MASQ-Project/Node/actions/runs/403501975 scroll down to the Artifact section. Successful workflows will have an archive of the results. I'm not sure how long they survive. I think it  looks to be everything captured in the `results` path but stored with the build output eliminating the need for a separate results repo.

kauri-hero commented:
looks like both @dnwiebe and @masqrauder, you have troubleshooted those integration test failures.
All 3 checks have passed, so are you happy to confirm the Review required?

dnwiebe commented on .idea/scopes/MASQ_Copyright.xml:
It looks like some kind of mechanism to automatically generate copyright comments, which I support; but I think you're right: it's automatically generated, and not very intelligently.

dnwiebe commented on .github/workflows/ci-matrix.yml:
That's beautiful! That's incredible--I love it. I never liked that results site; I'll be very happy to take it down.

One thing, though: can you modify CONTRIBUTING.md to reference your new way of doing things?

dnwiebe reviewed (approved):
If you'd like to merge it without making the changes to CONTRIBUTING.md, go ahead; I'll try to make the changes myself. You'd be better at it than I would, though, since you'd understand what you're talking about.

masqrauder commented on .github/workflows/ci-matrix.yml:
Sure I'll look at updating CONTRIBUTING.md.

kauri-hero commented:
Approved.
Reviewed Contributing and will add some more details with documentation branch I am working on for other readme files

kauri-hero commented:
wierd @masqrauder your one-line commit to the Contributing.md didn't make it over in the PR.
I'm updating this today with some other documentation updates

masqrauder committed:
Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

masqrauder committed:
Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

masqrauder committed:
GH-600: Handle expected batch, got single response

masqrauder committed:
GH-600: Handle expected batch, got single response

utkarshg6 committed:
GH-701: remove deprecated (September 2021) manifest keys from Cargo.toml files (#301)

bertllll committed:
GH-719: Replace BlockchainInterfaceClandestine with BlockchainInterfaceNull (#307)

* GH-719: Blockchain interfaces slightly modified, mainly naming. The clandestine was replaced with BlockchainInterfaceNull

* GH-719: review issues fixed

* GH-719: review two answered

* GH-719: review 3 answered

* GH-719: additional changes for review 3

* GH-719: web3 ABI reformated

---------

Co-authored-by: Bert <Bert@Bert.com>

masqrauder committed:
Resolve rebase issues

masqrauder committed:
Resolve rebase issues

utkarshg6 committed:
GH-691: Integrate `--min-hops` with `PersistentConfiguration` (#294)

* GH-691: database migration for 7 to 8

* GH-691: update CURRENT_SCHEMA_VERSION to 8

* GH-691: add test to set the min_hops_count inside PersistentConfiguration

* GH-691: fix tests in actor_system_factory.rs

* GH-691: add the ability to retrieve value of min_hops_count from the database

* GH-691: add tests for validating min hops count

* GH-691: test drive code for validating the min_hops_count value from the persistent configuration

* GH-691: minor code improvements

* GH-691: remove second declaration of CURRENT_SCHEMA_VERSION

* GH-691: remove unnecessary comments

* GH-691: improve validate_database_min_hops_count()

* GH-691: fixed test cleanup_after_deceased_clients_integration

* GH-691: add todos to where we should set the min_hops_count

* GH-691: set min_hops_count value inside configure_database()

* GH-691: remove code for setting min_hops_count inside actor_system_factory.rs

* GH-691: set min_hops_count default value while initializing database

* GH-691: change how min_hops_count is handled inside make_neighborhood_config()

* GH-691: instead of defaulting min_hops_count inside clap, default it in db

* GH-691: feed min_hops_count value inside ConfigDaoNull

* GH-691: add test for checking whether the node panics when the min_hops_count value is not found

* GH-691: remove clippy warnings

* GH-691: remove redundant code

* GH-691: more improvements to  validate_database_min_hops_count()

* GH-691: Review 1 (#300)

* GH-691: minor changes in persistent_configuration.rs

* GH-691: Display for Hops can be simplified

* feat: panic when value of min_hops_count is not present in db

* GH-591: remove setting a result for set_min_hops_count inside actor_system_factory.rs

* GH-691: improvements in migration_7_to_8.rs

* fix: the todo has been changed to a comment after discussion in the dev team

* GH-691: Review 2 (#302)

* GH-691: rename min_hops_count to min_hops everywhere

* GH-691: implement the computed_default()

* GH-691: implement the computed_default() for min hops

* GH-691: replace min hops value to the the one in db

* GH-691: rename DEFAULT_MIN_HOPS_COUNT and MIN_HOPS_COUNT_FOR_TEST to DEFAULT_MIN_HOPS and MIN_HOPS_FOR_TEST respectively

* GH-691: remove unnecessary test

* GH-691: Review 3 (#304)

* GH-691: convert panic into error log inside the setup_reporter.rs

* GH-691: use Default instead of Configured if the default min hops value is found in db

* GH-691: change debug to info in the validation

* GH-691: use AssertionsMessage for asserting when the validation does not need to replace the value

* GH-691: remove use of min_hops_params from the tests in Neighborhood

* GH-691: remove min_hops_params

* GH-691: fix typo in test name

* GH-691: improve the log statement

* GH-691: fix the remaining renames (#312)

utkarshg6 committed:
GH-727: Don't run actions when PR is in Draft Mode (#319)

masqrauder committed:
GH-600: Adapt to new Pokt message changes

masqrauder committed:
GH-600: Adapt to new Pokt message changes

masqrauder committed:
Use start_block in the event of an error

masqrauder committed:
Use start_block in the event of an error

masqrauder committed:
GH-600: Return blockchain scan error to update max_block_count

masqrauder committed:
GH-600: Return blockchain scan error to update max_block_count

bertllll committed:
GH-580: test for data_directory_from_context with opt dir and implementation … (#248)

* test for data_directory_from_context with opt dir and implementation in fn

* fixed naming and optimizing call for add_chain_specific_directories()

* add_chain_specific_directories to tests in  bootstrapper.rs, mod.rs, node_configuraton_standard.rs

* calculate_configured_setup returns None -> panicked at 'called  on a  value'

* formatting

* cleaning misstake

* commit changes - move to laptop

* fix tests failing on the chain specific data direcotry

* changes based on Berts review comments

* formatting

* remove todo causing crash

* half way to fix cleanup_after_deceased_clients_integration

* failing on creating log file in node_logs_panic_integration

* change joins for add_chain_specific_directories fn

* remove unused import

* add_chain_specific_directories moved to masq_lib::utils, introduced new function node_chain_specific_directory for multinode tests

* remove comments

* remove comments, replace joins wtih add_chain_specific_directories fn, test for chain specific directories uncompleted but passing

* Revert "remove unused import"

This reverts commit c0f205a2cd6c5ecb1d53e48cf2763ffacd1455f4.

* pull and resolve conflicts

* vec! with foreach for test server_initializer_collected_params_senses_when_user_specifies_data_directory_without_chain_specific_directory

* add more scenarios for test with custom data_directory_opt in test server_initializer_collected_params_senses_when_user_specifies_data_directory_without_chain_specific_directory

* continue of solving Bert's comments, start of modifying UI to show full path for user specified data-directory

* try to retrieve chain-name from inner.values

* chain_name collecting with map().collect, value_value collecting with match

* move match for value_value to separate fn to clar dump_setup

* chainge test for right fn calls, added message about changed data-directory to match chain parameter

* change of chain_specific_data_directory to add alway chain_specific_dir to path set by user, so chain dir could be duplicated, finalize test for masq UI to check various UiSetupResponsValueStatus

* new integration test for specific data_directory ensure_data_directory_has_specific_chain_direcotry_within

* fix Default arm for directory path enrichment with chain specific directory

* integration test to ensure setup command do the right job to add chain specific directory

* change data-directory, comparsion and panic messages in chain_specific_directory_test_integration test, remove comments inn setup_command

* fix formatting for expected in chain_specific_directory test, remove unused import from setup_command

* finalize test for setup_reporter calculate_fundamentals_with_chain_specific_dir

* commented code to test in linux

* compilable version to test in linux

* move changes to desktop

* for bert from my linux machine

* GH-580: showing the right direction

* fixing more tests

* GH-580: ongoing repairment in setup_reporter

* GH-580: fixed data directory niches that were causing a lot of troubles

* remove unvanted println

* formatting

* fixed test calculate_setup_with_chain_specific_dir to pass on mac, resolved last Berts comments

* remove unused imports and chain from mock node

* remove unused code from setup_command

* resolving review comments - add closure for retriving data-dir and chain

* resolving comments: cleaning setup_command tests, cleaning setup_reporter:get_modified_setup, fix misspe<lls

* try to send less data to determine_setup_value_of_set_data_directory fn

* remove unused data from SetupReporterReal::determine_setup_value_of_set_data_directory

* formatting

* fixed test for get_modified_setup::get_modified_setup_user_specified_data_directory_depends_on_new_chain_on_success

* resolved review comments on ensure_data_directory_has_specific_chain_directory_within_integration test

* split of test calculate_setup_with_chain_specific_dir into two separate tests

* resolving review comments, renaming, optimization, fixing tests

* resolving review comments - mainly renaming

* Half way to get rid of DirsWrapper from go fn in DumpConfigRunnerReal

* remove DirsWrapper from arguments to DumpConfigRunner for DumpConfigRunnerReal

* fixed windows tests for determine_fundamentals

* renaming functions, removing TODOs and removing unnecesary attributes from test

* fixed node_chain_specific_data_directory and impl MASQRealNode plus try to fix node_creates_log_file_with_heading_integration - continue next

* fixed node_creates_log_file_with_heading_integration

* remove unused import and debug print, tests failing in ci/all.sh but passing when run from cmd line

* fix paths in tests for windows: setup_reporter

* fix windows paths for test server_initializer_collected_params_senses_when_user_specifies_data_directory_without_chain_specific_directory

* fix setup_command and broadcast_handler tests: added missing data-directory and chain parameters to config

* setup_command removign ropsten and correcting UiResponseValue, broadcast_handler: removing unwanted piece of code, shared_schema: DATA_DIRECTORY_HELP, setup_reporter: fix test

* correction of DATA_DIRECTORY_HELP

* lazy_static for STANDARD_ARG_LIST in shared_schema

* separate help for --data-directory in Node, Config Dump and UI

* remove duplication of data-directory clap app argument, add dirs to masq_lib to compute static constant DATA_DIRECTORY_DAEMON_HELP, fix broadcast_hadnler test

* move data-directory back to shared schema::shared_app::head

* remove typo misstake

* apply Dans comments on review 6, put code in the order, formatting and moving code thru directories and files

* remove unused import

* resolve DATA_DIRECTORY_DAEMON_HELP for masq UI

---------

Co-authored-by: Vojtěch Parkán <vojtechparkan@Vojtechs-iMac.home>
Co-authored-by: Vojtěch Parkán <vojtechparkan@Vojtechs-iMac.local>
Co-authored-by: Vojtěch Parkán <czarte@gmail.com>
Co-authored-by: Bert <Bert@Bert.com>

bertllll committed:
Little renamings (#333)

* little renaiming

* little renaming: formatting

---------

Co-authored-by: Bert <Bert@Bert.com>

masqrauder committed:
GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

masqrauder committed:
GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

utkarshg6 committed:
GH-745: Macro for creating HashMap (#337)

* macros-experiment: add hashmap!()

* GH-745: gitignore files

* GH-745: untrack file

* GH-745: trigger actions

Commit:
GH-672: mostly done; just a few details left

Commit:
GH-672: abondoning enhancement of blockchain interface initializer...leaving as it is

Commit:
GH-672: unnoticable change for the augly low level tests in actor_system_factory

Commit:
GH-672: blockchain_bridge_shared_test_simplified

Commit:
GH-672: ready for review 5

dnwiebe commented on masq_lib/src/utils.rs:
You thought the names here could use another going-over.

dnwiebe commented on node/src/accountant/scanners/mid_scan_msg_handling/payable_scanner/blockchain_agent.rs:
Is it reasonable to rely on the `#[cfg(test)]` inside `as_any_in_trait` instead of supplying another level here?

dnwiebe commented on node/src/accountant/scanners/mid_scan_msg_handling/payable_scanner/test_utils.rs:
Commented-out code

dnwiebe commented on node/src/accountant/scanners/mod.rs:
We spent some time talking about this in person.

dnwiebe commented on node/src/accountant/scanners/test_utils.rs:
If we can generify the obfuscation, this might go away. If we can't, it might bear a second look anyway, just because the triple layer of calls on the stack seems a little clumsy.

dnwiebe commented on node/src/blockchain/blockchain_bridge.rs:
Should be named `shared_tests` instead

dnwiebe commented on node/src/blockchain/blockchain_bridge.rs:
Idiomatic abbreviation for "receiver" is "rx," not "rv"

dnwiebe commented on node/src/blockchain/blockchain_bridge.rs:
Can you add a comment that the purpose of this assertion is to prove that the `BlockchainBridge` is being supplied with the correct `PersistentConfiguration`?

dnwiebe commented on node/src/blockchain/blockchain_bridge.rs:
You thought this name could be better. Me too: it tests two things, not one.

dnwiebe commented on node/src/blockchain/blockchain_bridge.rs:
This seems really brittle to me. Can you do some more intelligent parsing on this value? Is it comma-separated, for example?

dnwiebe commented on node/src/blockchain/blockchain_bridge.rs:
If you moved the stop-the-system call to just before this line, I think it would work, and if it did, all the System-related stuff would be in the same place.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/rpc_helpers_web3.rs:
If you could represent this in hexadecimal, it'd be easier to match up with the Arrange part of the test.

dnwiebe commented on node/src/blockchain/blockchain_interface/blockchain_interface_web3/rpc_helpers_web3.rs:
Again, hex would be nicer.

dnwiebe commented on node/src/accountant/mod.rs:
You could probably move this to just before `assert_eq!(system.run(), 0);`.

masqrauder committed:
GH-600: Add test for scan time out

masqrauder committed:
GH-600: Add test for scan time out

masqrauder committed:
GH-600: Revert scanner changes

masqrauder committed:
GH-600: Revert scanner changes

utkarshg6 committed:
GH-692:  Allow UI to configure --min-hops during runtime (#316)

* GH-692: ignore file dbnavigator.xml

* GH-692: add tests for command execution of set-configuration

* GH-692: revert the version number

* GH-692: add test for the error case

* GH-692: improve tests to work on macos

* GH-692: improve tests to work on macos

* GH-692: ignore dbnavigator.xml

* GH-692: remove the redundant gitignore line

* GH-692: add tests for setting min_hops using set-configuration

* GH-692: add TODOs

* GH-692: migrate the Consuming Wallet change and Password Change to a new SetConfigurationMessage

* GH-692: update field names inside SetConfigurationMessage and variants name inside ConfigurationChange

* GH-692: update field names inside SetConfigurationMessage and variants name inside ConfigurationChange

* GH-692: write test can_update_min_hops_with_set_configuration_msg

* GH-692: set min hops value in db inside node_configurator/configurator.rs

* GH-692: only change value of min hops in neighborhood using SetConfigurationMessage

* GH-691: write more tests for setting min hops inside the node_configurator/configurator.rs

* GH-692: an attempt to remove the SetConsumingWalletMessage

* GH-692: remove SetConsumingWalletMessage from all the other places because it's a dead code

* GH-692: remove the SetConsumingWalletMessage completely

* GH-692: remove NewPasswordMessage

* GH-692: replace NewPasswordMessage with SetConfigurationMessage

* GH-692: add test for UpdateNewPassword inside configurator.rs

* GH-692: write more tests for the configurator.rs

* GH-692: improve test handle_set_configuration_handles_failure_on_min_hops_database_issue

* GH-692: change name of message to ConfigurationChangeMessage

* GH-691: implement calculate_db_patch_size()

* GH-692: add the fn to set_min_hops_and_db_patch_size()

* GH-692: set db_patch_size while setting min_hops

* GH-692: completely remove NewPasswordMessage

* GH-692: drive code for setting up patch size inside NeighborhoodMetadata

* GH-692: use db_patch_size or patch_size instead of min_hops while computing patch

* GH-692: remove redundant functions

* GH-692: formatting changes

* GH-692: ignore the stray test - ci/all.sh is passing

* GH-692: migrate min_hops test to a new file in multinode tests

* GH-692: attempt to write multinode test

* GH-692: why is this working? and not the previous version

* GH-692: improve multinode test

* GH-692: test drive the RouteQuery after the min hops has been changed

* GH-692: refactor the neighborhood/mod.rs file

* GH-692: min hops multinode test is passing

* GH-692: improve multinode test with better comments and refactoring

* GH-692: improve logging inside Neighborhood and Configurator

* GH-692: another formatting changes

* GH-692: allow overall connection stage to regress

* GH-692: regress stage of OverallConnectionStatus and send a msg to the UI

* GH-692: assert that the ui_gateway received a message for regression

* GH-692
: remove unnecessary warnings

* GH-692: test hardening

* GH-692: reorder msg inside recorder.rs

* GH-692: minor improvements in multinode tests

* GH-692: rename file

* GH-692: few fn renames

* GH-692: trigger actions

* GH-692: remove unused import from multinode test and increase timeout

* GH-692: regress ocs stage only if the initial stage is RouteFound

* GH-692: Review 1 (#334)

* GH-692: add the public constant DEFAULT_TRANSACT_TIMEOUT_MILLIS for the production code

* GH-692: add another constant for the tests

* GH-692: reduce total number of nodes for the min hops data routing test

* GH-692: change the comment for the multinode test

* GH-692: replace the constant for db_patch_size and use the different name everywhere

* GH-692: put code back for NewPasswordMessage

* GH-692: minor changes

* GH-692: add error logging when the UiSetConfigurationRequest causes an error

* GH-692: test fix and remove todo

* GH-692: fix test set_configuration_command_throws_err_for_invalid_arg

* GH-692: remove UpdateNewPassword variant

* GH-692: add better error msg in case the assert statement fails

* GH-692: fix using --gas-price and --min-hops during runtime without any value

* GH-692: required is actually not required

* GH-692: set_configurationify() changes min_values to 1 instead of 0

* GH-692: formatting

Commit:
GH-672: interim commit

Commit:
GH-672: finished review 5

Commit:
GH-672: merge with master

Commit:
GH-672: review 6 answered

Commit:
GH-672: taking over code from master + small fixes

Commit:
GH-672: fixes in multinode tests

Commit:
GH-672: little cosmetics

bertllll committed:
GH-751: Hot fix for wrong books keeping at transactions confirmation  (#349)

* GH-751: done

* GH-751: answer to the review

---------

Co-authored-by: Bert <Bert@Bert.com>

Commit:
GH-672: merge with GH-600 completed

Commit:
GH-672: renamings and moves in test code

Commit:
GH-672: rpc halpers into lower level interface

masqrauder committed:
GH-600: Address review feedback

Commit:
GH-672: incorporated new changes from GH-600

Commit:
GH-672: ending a big rearrangement; stable with all the tests again

Commit:
GH-672: more refactoring

Commit:
GH-672: missing import

Commit:
GH-672: review 7 answered

masqrauder committed:
GH-600: Add get_block_number (#193)

* Squashing:
GH-600: Add get_block_number
GH-600: Address PR feedback
GH-585: Experiment with adapting test for platform specific error codes
GH-585: Provide Windows platform specific error code and message
GH-600: Rebase master into branch

* GH-600: Handle expected batch, got single response

* Resolve rebase issues

* GH-600: Adapt to new Pokt message changes

* Use start_block in the event of an error

* GH-600: Return blockchain scan error to update max_block_count

* GH-600: Resolve blockchain scanning

* Starts Accountant
* Adds Receivables scan timeout (may become a configuration parameter in another card)
* Stores `max_block_count` in the config table when processing error messages providing a limit
* Adds Max Block Count to configuration response
* Increments CURRENT_SCHEMA_VERSION
* Adds schema migration for `max_block_count`
* Improves blockchain scan error handling and logging

* GH-600: Add test for scan time out

* GH-600: Revert scanner changes

* GH-600: Address review feedback

* GH-600: Address review 6 feedback

* GH-600: Use From traits to be concise

Commit:
GH-672: merged master in

Commit:
GH-672: removed a not suitable comment

Commit:
GH-672: review 8 answered

Commit:
GH-672: lower interface renames
